### PR TITLE
MOB-18/23 Add Restart/GameOver screen functionality

### PR DESCRIPTION
**Pull request by sayeeem**
Restart Screen:
image

GameOver Screen:
image

NOTE: We can easily change how we want the look and feel of the screens, pushing this so we have the basic functionality on main

### CODE REVIEW

File: src/commonMain/kotlin/main.kt [hunk 1]
```diff
@@ -245,7 +245,47 @@ suspend fun main() = Korge(width = 480, height = 800, title = "2048", bgcolor =
 
 }
 
+fun Container.showGameOver(onGameover: () -> Unit) = container {
```
Comment (MarkGaskin) [inline src/commonMain/kotlin/main.kt:248]: Nit-pick but my IDE complains about `onGameover` vs `onGameOver`

File: src/commonMain/kotlin/main.kt [hunk 2]
```diff
@@ -245,7 +245,47 @@ suspend fun main() = Korge(width = 480, height = 800, title = "2048", bgcolor =
...
+	Napier.w("Showing GameOver Container...")
+	fun restart() {
+		this@container.removeFromParent()
+		onGameover()
+	}
+	position(leftIndent, topIndent)
+
+	var bgGameOverContainer = roundRect(435.0, 435.0, 5.0, fill = Colors["#aaa6a4cc"])
```
Comment (MarkGaskin) [inline src/commonMain/kotlin/main.kt:256]: Can be `val` instead of `var`, since it's never updated

File: src/commonMain/kotlin/main.kt [hunk 3]
```diff
@@ -245,7 +245,47 @@ suspend fun main() = Korge(width = 480, height = 800, title = "2048", bgcolor =
...
+	val bgGameOverText = roundRect(cellSize * 6.5, cellSize * 4.0, 25.0, fill = Colors["#aaa6a4"]) {
+		centerXOn(bgGameOverContainer)
+		y -= -110
+	}
+	val bgTryagainText = roundRect(cellSize * 4.0, cellSize * 1.0, 20.0, fill = Colors["#aaa6a4"], stroke = Colors.BLACK, strokeThickness = 1.5) {
```
Comment (MarkGaskin) [inline src/commonMain/kotlin/main.kt:261]: More nit-picking: `bgTryAgainText`

File: src/commonMain/kotlin/main.kt [hunk 4]
```diff
@@ -245,7 +245,47 @@ suspend fun main() = Korge(width = 480, height = 800, title = "2048", bgcolor =
 
 }
 
+fun Container.showGameOver(onGameover: () -> Unit) = container {
+	Napier.w("Showing GameOver Container...")
```
Comment (MarkGaskin) [inline src/commonMain/kotlin/main.kt:249]: I've been bad about this, but `Napier.w` is for warnings. Probably better to use debug (`Napier.d`) here instead and other log statements

**Comment by MarkGaskin** [comment]
Need to add a guard to bomb1Container / bomb2Container's onClick:
```
		onClick {
			if(bomb2Loaded.value && !showingRestart ) {
				bomb2Selected = !bomb2Selected
				animateBombSelection(this, bomb2Selected)
			}
		}
```

**Comment by sayeeem** [comment]
> && !showingRestart

added!

File: src/commonMain/kotlin/main.kt (hunk 1 above)
Comment (MarkGaskin) [reply src/commonMain/kotlin/main.kt:248]: Test

**Pull request merged**